stablehlo/reference/Interpreter.cpp

```diff
@@ -35,142 +36,174 @@ llvm::Expected<SmallVector<Tensor>> eval(func::FuncOp func,
     return invalidArgument("Expected one region in func %s",
                            func.getName().str().c_str());
 
-  if (!func.getBody().hasOneBlock())
-    return invalidArgument("Expected one block in func %s",
-                           func.getName().str().c_str());
-
   Block &block = func.front();
   if (block.getNumArguments() != args.size())
     return invalidArgument(
         "Expected same amount of func arguments in %s "
         "and runtime arguments (%d)",
         func.getName().str().c_str(), args.size());
```

Comment:
I kept this check separate from the similar check inside `evalRegion` mainly because 

1. The check in function is a input error (as the input `Tensors` are provided by the user), whereas the check in `evalRegion` is a fatal error as it signifies something is wrong which the implementation of the interpreter. 

2. The diagnostic message for this check is more informative than what we have in `evalRegion`, which is

```
  Block &block = region.front();
  if (block.getArguments().size() != runtimeArgs.size())
    report_fatal_error(invalidArgument(
        "Expected same amount of block arguments and runtime arguments (%d)",
        runtimeArgs.size()));
  ...
```

In order to implement the above differences, having two copies seems a less overhead solution to me.
 
Let me know if the above justifies keeping those checks are separate. Otherwise happy to de-duplicate.

Reply:
After moving `evalRegion` to `Ops.*`, chose to removed the   `Interpreter.* files`  (as part of comment https://github.com/openxla/stablehlo/pull/1093#discussion_r1103221122). With that the the above check and call to `evalRegion` are hoisted to the interpreter [driver](https://github.com/openxla/stablehlo/blob/main/stablehlo/tools/StablehloInterpreterMain.cpp#L32). As the `arg` passed on to `eval(func::FuncOp)`  were empty, the above check will be trivially  folded.